extensions/builder.md

```diff
@@ -0,0 +1,180 @@
+| `CNB_LAYERS_DIR`   | The directory to create and store `layers` in the build environment               | `/layers`    |
+| `CNB_PLATFORM_DIR` | The directory to create and store platform focused files in the build environment | `/platform` |
+
+The following variables MAY be set on the builder (through the image config's `Env` field):
+
+| Env Variable           | Description                            | Default |
+| ---------------------- | -------------------------------------- | ---- |
+| `SERVICE_BINDING_ROOT` | The directory where services are bound | - |
+| `CNB_BUILDPACKS_DIR` | The directory where buildpacks are located | `/cnb/buildpacks` |
+
+### Labels
+The following labels MUST be set in the builder environment (through the image config's `Labels` field):
+
+| Label                             | Description                                                                                                              |
+| --------------------------------- | ------------------------------------------------------------------------------------------------------------------------ |
+| `io.buildpacks.builder.api`       | The [Builder API Version](#builder-api-version) this builder implements                                                  |
+| `io.buildpacks.builder.metadata`  | A JSON object representing [Builder Metadata](#iobuildpacksbuildermetadata)                                              |
+| `io.buildpacks.buildpack.order`   | A JSON object representing the [order of the buildpacks stored on the builder](#iobuildpacksbuildpackorder)              |
+| `io.buildpacks.buildpack.layers`  | A JSON object representing the [buildpack layers](#iobuildpacksbuildpacklayers)                                          |
+| `io.buildpacks.lifecycle.version` | A string, representing the version of the lifecycle stored in the builder                                                |
+| `io.buildpacks.lifecycle.apis`    | A JSON object representing the [lifecycle APIs](#iobuildpackslifecycleapis) the lifecycle stored in the builder supports |
+
+## Data Format
+### Labels
+#### `io.buildpacks.builder.metadata` (JSON)
+
+```json
+{
+  "description": "<description>",
+  "stack": {
+    "runImage": {
+      "image": "<run image>",
+      "mirrors": [
+        "<run image mirror>"
+      ]
+    }
+  },
+  "buildpacks": [
+    {
+      "id": "<buildpack ID>",
+	  "version": "<buildpack version>",
+	  "homepage": "<buildpack homepage>"
+	}
+  ],
+  "createdBy": {
+    "name": "<tool name>",
+    "version": "<tool version>"
+  }
+}
+```
+
+The `createdBy` metadata is meant to contain the name and version of the tool that created the builder. 
+
+#### `io.buildpacks.buildpack.order` (JSON)
+
+```json
+[
+	{
+	  "group":
+		[
+		  {
+			"id": "<buildpack ID>",
+			"version": "<buildpack version>",
+			"optional": false
+		  }
+		]
+	}
+]
+```
+
+#### `io.buildpacks.buildpack.layers` (JSON)
```

Comment:
I noticed that we don't define this label in the distribution spec even though we use it in pack in order to create this label on builders. This feels like an oversight so I created an issue to address it https://github.com/pivotal/kpack/search?q=io.buildpacks.buildpack.layers

Reply:
Is that issue supposed to be on kpack?

Reply:
whoops, thanks